components/kyma-environment-broker/internal/process/upgrade_cluster/initialisation.go

```diff
@@ -155,7 +155,12 @@ func (s *InitialisationStep) performRuntimeTasks(step int, operation internal.Up
 
 	switch step {
 	case UpgradeInitSteps:
-		if hasMonitors && !inMaintenance {
+		if s.evaluationManager.IsMaintenanceModeDisabled() {
```

Comment:
I thnik the performRuntimeTask plays only with AvS and whole code of that method can be skipped when maintanance mode is disabled. Let's discuss

Reply:
We discussed the matter and decided to leave the change as it is.